### PR TITLE
[MISC] Move core/common_tuple.hpp -> core/tuple/common_tuple.hpp

### PR DESCRIPTION
**Pull request by eaasna**
Part of [seqan/product_backlog#160](https://github.com/seqan/product_backlog/issues/160)

### CODE REVIEW

**Comment by smehringer** [comment]
Nice! :) All tests pass :+1: You can now request seqan/team on the right hand side, such that the first review round starts.

**Comment by smehringer** [comment]
Rene and I are always doing the second review :) Other team members are doing the first. You can literally request `seqan/team` and it will randomly select one of the first review team members. Just search for `team` in the review box next time.

File: include/seqan3/core/common_tuple.hpp [hunk 1]
```diff
@@ -6,35 +6,13 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
- * \brief Provides seqan3::common_tuple and seqan3::common_pair.
+ * \brief [DEPRECATED] Provides seqan3::common_tuple and seqan3::common_pair.
+ * \deprecated This header is deprecated and will be removed in SeqAn-3.1.
+ *             Please \#include <seqan3/core/tuple/common_tuple.hpp> instead.
```
Comment (MitraDarja) [inline include/seqan3/core/common_tuple.hpp:11]:
```suggestion
* \deprecated This header will be removed in 3.1.0; Please \#include <seqan3/core/tuple/common_tuple.hpp> instead.
```

File: include/seqan3/core/common_tuple.hpp [hunk 2]
```diff
@@ -6,35 +6,13 @@
...
- *        output iterators.
- *
- * \details
- *
- * Alias definition of the ranges::common_pair.
- */
-using SEQAN3_DOXYGEN_ONLY(common_pair =) ::ranges::common_pair;
-
-}  // namespace seqan3
+SEQAN3_DEPRECATED_HEADER("This header is deprecated and will be removed in SeqAn-3.1. "
```
Comment (MitraDarja) [inline include/seqan3/core/common_tuple.hpp:17]:
```suggestion
SEQAN3_DEPRECATED_HEADER("This header is deprecated and will be removed in SeqAn-3.1.0. "
```

**Review by MitraDarja — CHANGES_REQUESTED** [review]
Looks good, well done! 👍 
Just two small things then it can go to the second review. Just request me again, once you have added the changes.

File: include/seqan3/core/common_tuple.hpp [hunk 3]
```diff
@@ -6,35 +6,12 @@
...
- *
- * \details
- *
- * Alias definition of the ranges::common_pair.
- */
-using SEQAN3_DOXYGEN_ONLY(common_pair =) ::ranges::common_pair;
-
-}  // namespace seqan3
+SEQAN3_DEPRECATED_HEADER("This header is deprecated and will be removed in SeqAn-3.1.0 "
+                         "Please #include <seqan3/core/tuple/common_tuple.hpp> instead.")
```
Comment (MitraDarja) [inline include/seqan3/core/common_tuple.hpp:17]: There is a new line missing in the end.

**Review by MitraDarja — CHANGES_REQUESTED** [review]
Just two more things. There is a merge conflict now. Can you solve it? :)

File: include/seqan3/core/common_tuple.hpp (hunk 3 above)
Comment (eaasna) [reply include/seqan3/core/common_tuple.hpp:17]: Oh, thanks for noticing :)

**Review by MitraDarja — APPROVED** [review]
Looks good. 👍

**Comment by marehr** [comment]
@smehringer Can you review this PR? It is waiting some days already :)

**Comment by smehringer** [comment]
whoope I really thought I had reviewed that one. Thanks for the ping

File: include/seqan3/core/common_tuple.hpp [hunk 4]
```diff
@@ -6,35 +6,13 @@
...
+ * \deprecated This header will be removed in 3.1.0; Please \#include <seqan3/core/tuple/common_tuple.hpp> instead.
  * \author Rene Rahn <rene.rahn AT fu-berlin.de>
  */
 
-#pragma once
+#include <seqan3/core/tuple/common_tuple.hpp>
 
-#include <range/v3/utility/common_tuple.hpp>
+SEQAN3_DEPRECATED_HEADER("This header is deprecated and will be removed in SeqAn-3.1.0 "
+                         "Please #include <seqan3/core/tuple/common_tuple.hpp> instead.")
```
Comment (smehringer) [inline include/seqan3/core/common_tuple.hpp:17]:
```suggestion
SEQAN3_DEPRECATED_HEADER(
"This header is deprecated and will be removed in SeqAn-3.1.0 Please #include <seqan3/core/tuple/common_tuple.hpp> instead.")
```

**Review by smehringer — APPROVED** [review]
only a tiny thing

**Pull request merged**